### PR TITLE
Correct UTF8 detection which was being detected as ISO-8859-1

### PR DESCRIPTION
**Pull request by saurabh95**

### CODE REVIEW

File: appshell/appshell_extensions_mac.mm [hunk 1]
```diff
@@ -654,46 +654,55 @@ int32 ReadFile(ExtensionString filename, ExtensionString& encoding, std::string&
...
+    if (encoding == "UTF-8") {
+        enc = NSUTF8StringEncoding;
+        NSError* NSerror = nil;
+        fileContents = [NSString stringWithContentsOfFile:path encoding:enc error:&NSerror];
+    }
+    
+    if (fileContents)
+    {
+        contents = [fileContents UTF8String];
+        CheckForUTF8BOM(filename, preserveBOM);
```
Comment (saurabh95) [inline appshell/appshell_extensions_mac.mm]: This is the only function call added new in this PR, rest is just revert of my last PR https://github.com/adobe/brackets-shell/pull/610/files#diff-dbb405241ed6688d3fab6d77d1202508

File: appshell/appshell_extensions_platform.cpp [hunk 2]
```diff
@@ -92,8 +93,23 @@ void DecodeContents(std::string &contents, const std::string& encoding) {
...
-    if (contents.length() >= 3 && contents.substr(0,3) == UTF8_BOM) {
-        contents.erase(0,3);
+    if (contents.length() >= 3 && contents.substr(0, 3) == UTF8_BOM) {
+        preserveBOM = true;
+        contents.erase(0, 3);
+    }
+}
+
+void CheckForUTF8BOM(const std::string& filename, bool& preserveBOM) {
+    std::ifstream file(filename.c_str());
```
Comment (nethip) [inline appshell/appshell_extensions_platform.cpp]: Wrap this piece of code in a try catch block

**Comment by nethip** [comment]
@saurabh95 Add sufficient comments about BOM detection. Otherwise the code looks fine.

**Comment by saurabh95** [comment]
Sure

**Pull request merged**